gh-141510: Update Whats News for frozendict (#144961)
python/cpython: Doc/whatsnew/3.15.rst

:pep:`814`: Add frozendict built-in type
----------------------------------------

A new public immutable type, :class:`frozendict`, is added to the :mod:`builtins` module.
It does not allow modification after creation. A ``frozendict`` is not a subclass of ``dict``;
it inherits directly from ``object``. A ``frozendict`` is hashable (i.e., ``hash(frozendict)`` works)
as long as all of its keys and values are hashable.

>>> a = frozendict(x=1, y=2)
>>> a
frozendict({'x': 1, 'y': 2})
>>> a['z'] = 3
Traceback (most recent call last):
  File "<python-input-2>", line 1, in <module>
    a['z'] = 3
    ~^^^^^
TypeError: 'frozendict' object does not support item assignment
>>> b = frozendict(y=2, x=1)
>>> hash(a) == hash(b)
True
>>> a == b

> True

Reviewer: ```suggestion
For example::

    >>> a = frozendict(x=1, y=2)
    >>> a
    frozendict({'x': 1, 'y': 2})
    >>> a['z'] = 3
    Traceback (most recent call last):
      File "<python-input-2>", line 1, in <module>
        a['z'] = 3
        ~^^^^^
    TypeError: 'frozendict' object does not support item assignment
    >>> b = frozendict(y=2, x=1)
    >>> hash(a) == hash(b)
    True
    >>> a == b
    True
```

:pep:`814`: Add frozendict built-in type
----------------------------------------

> A new public immutable type, :class:`frozendict`, is added to the :mod:`builtins` module.

Reviewer 2: ```suggestion
A new :term:`immutable` type, :class:`frozendict`, is added to the :mod:`builtins` module.
```
IMO public is redundant here.

It does not allow modification after creation. A ``frozendict`` is not a subclass of ``dict``;

> it inherits directly from ``object``. A ``frozendict`` is hashable (i.e., ``hash(frozendict)`` works)

Reviewer 2: Please don't use latin abbreviations https://devguide.python.org/documentation/style-guide/#use-simple-language.

> it inherits directly from ``object``. A ``frozendict`` is hashable (i.e., ``hash(frozendict)`` works)

Reviewer 2: These lines should be wrapped.

> it inherits directly from ``object``. A ``frozendict`` is hashable (i.e., ``hash(frozendict)`` works)

Reviewer 2: ```suggestion
It does not allow modification after creation. A ``frozendict`` is not a subclass of ``dict``;
it inherits directly from ``object``. A ``frozendict`` is :term:`hashable`
```

as long as all of its keys and values are hashable.

> 

Reviewer: Ah, you may add that it preserves insertion order, but that comparison doesn't take the order in account (your example proves it).
